### PR TITLE
add animal-sniffer

### PR DESCRIPTION
**Pull request by kwin**
only declare versions in pluginMgmt

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -410,6 +411,16 @@
                     <artifactId>maven-source-plugin</artifactId>
                     <version>3.0.1</version>
                 </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-enforcer-plugin</artifactId>
+                    <version>3.0.0-M2</version>
```
Comment (tomasalmeida) [inline pom.xml:417]: A new version is available (3.0.0-M3), should this plugin be updated?

File: pom.xml [hunk 2]
```diff
@@ -410,6 +411,16 @@
...
+                </plugin>
+                <plugin>
+                    <groupId>org.sonatype.plugins</groupId>
+                    <artifactId>nexus-staging-maven-plugin</artifactId>
+                    <version>1.6.3</version>
```
Comment (tomasalmeida) [inline pom.xml:422]: A new version is available (1.6.8), should this plugin be updated?

File: pom.xml [hunk 3]
```diff
@@ -489,14 +505,34 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>
-                    <source>1.7</source>
-                    <target>1.7</target>
+                    <source>1.${java.target.version}</source>
+                    <target>1.${java.target.version}</target>
```
Comment (tomasalmeida) [inline pom.xml:509]: in order to be able to upgrade to junit5, is it a good idea to use different jdk for unit testing and code compilation (as described here -> https://gist.github.com/aslakknutsen/9648594)?

**Review by tomasalmeida — APPROVED** [review]

**Comment by kwin** [comment]
We will do plugin upgrades with version 3.0 (in the context of #400).

**Pull request merged**